### PR TITLE
Improve gcp scripts to make config file optional

### PR DESCRIPTION
**Pull request by chrismaree**
This PR makes specifying environment variables config option with the `UpdateBotGCP.sh` and `PushBotHotFix.sh` scripts. If no configs are specified then the GCP instances will keep their previous configs.

### CODE REVIEW

File: scripts/bot-deployment/PushBotHotFix.sh [hunk 1]
```diff
@@ -1,22 +1,37 @@
 #!/bin/bash
 set -e
 
-if [ $# -ne 1 ]; then
-    echo "Incorrect number of arguments supplied! First and only argument is bot's name."
-    echo "example: ./PushBotHotFix.sh ethbtc-monitor-bot"
+if [ $# -eq 0 ]; then
+    echo "Incorrect number of arguments supplied! First and only argument is bot's name. Optionally include a config file"
```
Comment (nicholaspai) [inline scripts/bot-deployment/PushBotHotFix.sh:5]: nit: Change to "First and only required argument is bot's name".

File: scripts/bot-deployment/UpdateBotGCP.sh [hunk 2]
```diff
@@ -1,17 +1,31 @@
...
-echo "Using ENV file:" $2
-gcloud compute instances update-container $1 \
-    --container-image docker.io/umaprotocol/protocol:latest \
-    --container-env-file $2 \
-    --zone us-central1-a \
-    --container-restart-policy on-failure \
-    --container-stdin
+if [ $# -eq 1 ]; then
+    echo "Updating" $1
+    echo "Using no new enviroment variable. Keeping current configuration"
```
Comment (mrice32) [inline scripts/bot-deployment/UpdateBotGCP.sh:13]:
```suggestion
    echo "No configuration file provided. Keeping current configuration"
```

File: scripts/bot-deployment/UpdateBotGCP.sh [hunk 3]
```diff
@@ -1,17 +1,31 @@
 #!/bin/bash
 set -e
 
-if [ $# -ne 2 ]; then
-    echo "Incorrect number of arguments supplied! First argument is bot's name. Second argument is path to config file."
-    echo "example: ./UpdateBotGCP.sh ethbtc-monitor-bot ./ethbtc-monitor-bot-env.txt"
+if [ $# -eq 0 ]; then
+    echo "Incorrect number of arguments supplied! First argument is bot's name. Second optional argument is path to config file."
```
Comment (mrice32) [inline scripts/bot-deployment/UpdateBotGCP.sh:5]:
```suggestion
    echo "Incorrect number of arguments supplied! First argument is the bot's name. Second argument (optional) is a config file with environment variables to supply to the bot."
```

File: scripts/bot-deployment/UpdateBotGCP.sh [hunk 4]
```diff
@@ -1,17 +1,31 @@
...
+    echo "Using no new enviroment variable. Keeping current configuration"
+    gcloud compute instances update-container $1 \
+        --container-image docker.io/umaprotocol/protocol:latest \
+        --zone us-central1-a \
+        --container-restart-policy on-failure \
+        --container-stdin
+
+fi
+
+if [ $# -eq 2 ]; then
```
Comment (mrice32) [inline scripts/bot-deployment/UpdateBotGCP.sh:22]: What happens if 3 inputs are provided?

File: scripts/bot-deployment/PushBotHotFix.sh [hunk 5]
```diff
@@ -1,22 +1,37 @@
...
 echo "Pushing docker container to docker hub..."
-gcloud compute instances update-container $1 \
-    --container-image docker.io/umaprotocol/protocol:hotfix \
-    --zone us-central1-a \
-    --container-restart-policy on-failure \
-    --container-stdin
\ No newline at end of file
+if [ $# -eq 1 ]; then
+    echo "Pushing hot fix to" $1
+    echo "Using no new enviroment variable. Keeping current configuration."
```
Comment (mrice32) [inline scripts/bot-deployment/PushBotHotFix.sh:20]:
```suggestion
    echo "No configuration file provided. Keeping current configuration"
```

File: scripts/bot-deployment/PushBotHotFix.sh [hunk 6]
```diff
@@ -1,22 +1,37 @@
 #!/bin/bash
 set -e
 
-if [ $# -ne 1 ]; then
-    echo "Incorrect number of arguments supplied! First and only argument is bot's name."
-    echo "example: ./PushBotHotFix.sh ethbtc-monitor-bot"
+if [ $# -eq 0 ]; then
+    echo "Incorrect number of arguments supplied! First and only required argument is bot's name. Optionally include a config file."
```
Comment (mrice32) [inline scripts/bot-deployment/PushBotHotFix.sh:5]:
```suggestion
    echo "Incorrect number of arguments supplied! First argument is the bot's name. Second argument (optional) is a config file with environment variables to supply to the bot."
```

File: scripts/bot-deployment/PushBotHotFix.sh [hunk 7]
```diff
@@ -1,22 +1,37 @@
...
+    echo "Pushing hot fix to" $1
+    echo "Using no new enviroment variable. Keeping current configuration."
+    gcloud compute instances update-container $1 \
+        --container-image docker.io/umaprotocol/protocol:hotfix \
+        --zone us-central1-a \
+        --container-restart-policy on-failure \
+        --container-stdin
+fi
+
+if [ $# -eq 2 ]; then
```
Comment (mrice32) [inline scripts/bot-deployment/PushBotHotFix.sh:28]: Same as below -- what happens if 3 arguments are provided?

File: scripts/bot-deployment/UpdateBotGCP.sh (hunk 4 above)
Comment (chrismaree) [reply scripts/bot-deployment/UpdateBotGCP.sh:22]: if you put in three then the configuration is wrong and nothing happens.

Comment (mrice32) [reply scripts/bot-deployment/UpdateBotGCP.sh:22]: Maybe we could add a quick `or` on the first if to also check against too many arguments.

https://unix.stackexchange.com/a/47585

**Pull request merged**